libpromises/evalfunction.c

```diff
@@ -8722,6 +8722,35 @@ static FnCallResult FnCallCFEngineCallers(EvalContext *ctx, ARG_UNUSED const Pol
     return FnReturnContainerNoCopy(callers);
 }
 
+static FnCallResult FnCallString(ARG_UNUSED EvalContext *ctx,
+                                 ARG_UNUSED const Policy *policy,
+                                 ARG_UNUSED const FnCall *fp,
+                                 const Rlist *finalargs)
+{
+    // Parser doesnt allow this to be true since STRING_ARGS has one argument
+    if(finalargs == NULL)
+    {
+        return FnFailure();
+    }
+
+    char *ret;
+    bool allocated = false;  // Does it imply heap allocation is present?
+    JsonElement* json = VarNameOrInlineToJson(ctx, fp, finalargs, false, &allocated);
```

Comment:
So if IIRC this would actually make `string("var_name")` return the string representation of the `$(var_name)` variable (if it exists). I'm not sure that's what we want here. I think `string("some_string")` should just be `"some_string"`.

Reply:
IOW, I don't think this function should be a collecting function.

Reply:
I tested it with collecting and it doesnt seem to do so, but true im not really sure COLLECTING is necessary. With NONE it works properly anyway